dragonfly/server.go

```diff
@@ -381,6 +379,13 @@ func (server *Server) loadWorld() {
 		server.log.Fatalf("error loading world: %v", err)
 	}
 	server.world.Provider(p)
+	if _, ok := server.world.Generator().(world.NopGenerator); ok {
+		server.world.SetGenerator(generator.Flat{})
+	}
+	if p.FirstLoad() {
+		server.world.SetSpawn(world.BlockPos{0, server.world.HighestBlock(0, 0), 0})
+	}
```

Comment:
Are you forgetting to set FirstLoad to false?

Reply:
No, FirstLoad is meant to be a way of checking if the level data existed before the the world was loaded, so as a way of checking if this is the first time the server is generating level data for the world.

Reply:
If the level.dat doesn't exist, default data is used and in it we set FirstLoad to true because we can assume that this is the first time the world data is being created. I could probably rename this to something less confusing.

Reply:
Okay that makes sense